src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -34,53 +35,470 @@ const ActorInit En_Bombf_InitVars = {
-    { COLTYPE_NONE, AT_ON | AT_TYPE_ALL, AC_NONE, OC1_NONE, OC2_NONE, COLSHAPE_JNTSPH, },
-    1, D_808AFB4C, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ALL,
+        AC_NONE,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_JNTSPH,
+    },
+    1,
+    sJntSphElementsInit,
 };
 
-#endif
+Vec3f D_808AFB80 = { 0.0f, 0.0f, 0.0f };
+Vec3f D_808AFB8C = { 0.0f, 0.1f, 0.0f };
+Vec3f D_808AFB98 = { 0.0f, 0.0f, 0.0f };
+Vec3f D_808AFBA4 = { 0.0f, 0.6f, 0.0f };
+
+Color_RGBA8 D_808AFBB0 = { 255, 255, 255, 255 };
+
+void EnBombf_SetupAction(EnBombf* this, EnBombfActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnBombf_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    f32 yOffset = 0.0f;
+    GlobalContext* globalCtx = globalCtx2;
+    EnBombf* this = THIS;
+
+    Actor_SetScale(thisx, 0.01f);
+    this->unk_1F8 = 1;
+
+    Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+    Collider_InitJntSph(globalCtx, &this->colliderJntSph);
+    Collider_SetCylinder(globalCtx, &this->colliderCylinder, thisx, &sCylinderInit);
+    Collider_SetJntSph(globalCtx, &this->colliderJntSph, thisx, &sJntSphInit, this->colliderJntSphElements);
+
+    if (ENBOMBF_GET(thisx) == ENBOMBF_0) {
+        yOffset = 1000.0f;
+    }
+
+    ActorShape_Init(&thisx->shape, yOffset, ActorShadow_DrawCircle, 12.0f);
+
+    thisx->focus.pos = thisx->world.pos;
+    thisx->colChkInfo.cylRadius = 10;
+    thisx->colChkInfo.cylHeight = 10;
+    thisx->targetMode = 0;
+
+    if (ENBOMBF_GET(thisx) == ENBOMBF_0) {
+        this->timer = 140;
+        this->unk_1FE = 15;
+        thisx->gravity = -1.5f;
+        func_800BC154(globalCtx, &globalCtx->actorCtx, thisx, 3);
+        thisx->colChkInfo.mass = 200;
+        thisx->flags &= ~ACTOR_FLAG_1;
+        EnBombf_SetupAction(this, func_808AEE3C);
+    } else {
+        thisx->colChkInfo.mass = MASS_IMMOVABLE;
+        this->unk_1FC = 1;
+        this->unk_204 = 1.0f;
+        func_808AEAB8(this, ENBOMBF_GET(thisx));
+    }
+
+    thisx->uncullZoneScale += 31000.0f;
+    thisx->uncullZoneForward += 31000.0f;
+}
+
+void EnBombf_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBombf* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    Collider_DestroyJntSph(globalCtx, &this->colliderJntSph);
+}
+
+void func_808AEAB8(EnBombf* this, s16 arg1) {
+    EnBombf_SetupAction(this, func_808AEAE0);
+}
+
+void func_808AEAE0(EnBombf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    EnBombf* bombf;
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad3;
+
+    if (this->unk_204 >= 1.0f) {
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            bombf = (EnBombf*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOMBF, this->actor.world.pos.x,
+                                          this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
```

Comment:
```suggestion
            bombf = (EnBombf*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOMBF, this->actor.world.pos.x,
                                          this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, ENBOMBF_0);
``` 
Since you already had a define for it and its a simple params (spawning something that has three different thing in params and trying to shift them and stuff that would be ugly and we havent decided how to do that (right?), but in this case its really simple)